test/e2e/framework/kcp.go

```diff
@@ -227,6 +234,15 @@ func (c *kcpServer) Run(opts ...RunOption) error {
 		cleanupCancel()
 		return err
 	}
+
+	c.t.Cleanup(func() {
+		// Ensure child process is killed on cleanup
+		err := cmd.Process.Kill()
```

Comment:
This is how I managed to ensure shutdown of a server running as a subprocess in #795